README.Rmd

```diff
@@ -31,3 +31,146 @@ You can install the development version from [GitHub](https://github.com/) with:
 # install.packages("devtools")
 devtools::install_github("szymanskir/hackeRnews")
 ```
+
+## Examples
+
+### Word cloud
+
+Get recent words used in job story titles to build awesome word cloud. See what words are used the most frequently!
+
+```{r, warning=FALSE, message=FALSE}
+library(hackeRnews)
+library(tidyverse)
+library(tidytext)
+library(stringr)
+library(dplyr)
+library(ggwordcloud)
+
+job_stories <- hackeRnews::get_latest_job_stories()
+
+# get titles, normalize used words, remove non alphabet characters
+title_words <- unlist(
+  lapply(job_stories, FUN=function(job_story) { job_story$title }) %>% 
+  str_replace_all('[^A-Z|a-z]', ' ') %>% 
+  str_to_upper() %>% 
+  str_replace_all('\\s\\s*', ' ') %>% 
+  str_split(' ')
+)
+
+# remove stop words
+data('stop_words')
+df <- data.frame(word=title_words, stringsAsFactors=FALSE) %>% 
+  filter(str_length(word) > 0 & !str_to_lower(word) %in% stop_words$word) %>% 
+  count(word)
+
+# add some random color
+df <- as.data.frame(df) %>% 
+  mutate(color=factor(sample(10,nrow(df), replace=TRUE)))
+
+
+word_cloud <- ggplot(df, aes(label=word, size=n, color=color)) + 
+  geom_text_wordcloud() + 
+  scale_size_area(max_size = 15)
+
+```
+
+<img src="man/figures/word_cloud.png"/>
+
+### Best stories (based on score)
+
+```{r, warning=FALSE, message=FALSE}
+library(hackeRnews)
+library(stringr)
+library(ggplot2)
+
+best_stories <- hackeRnews::get_best_stories(max_items=10)
+df <- data.frame(
+  title=unlist(lapply(best_stories, FUN=function(best_story) { str_wrap(best_story$title, 42) })),
+  score=unlist(lapply(best_stories, FUN=function(best_story) { best_story$score })),
+  stringsAsFactors=FALSE
+)
+
+df$title = factor(df$title, levels=df$title[order(df$score)])
+
+best_stories_plot <- ggplot(df, aes(x = title, y = score, label=score)) +
+  geom_col() +
+  geom_label() +
+  coord_flip() +
+  xlab('Story title') +
+  ylab('Score') +
+  ggtitle('Best stories')
+
+```
+
+<img src="man/figures/best_stories.png"/>
+
+### Sentiment analysis on two best stories
+
+```{r, warning=FALSE, message=FALSE}
+library(hackeRnews)
+library(tidyverse, tidytext, dplyr)
+
+best_stories <- hackeRnews::get_best_stories(2)
+
+# get comments
+get_comments_contents <- function(item){
+  if( is.null(item$kids)){
+    c(item$text)
+  } else {
+    kids <- hackeRnews::get_items_by_ids(item$kids)
+    c(
+      item$text,
+      lapply(kids, get_comments_contents)
+    )
+  }
+}
```

Comment:
It might be useful to add functions to retrieves all items recursively maybe as an option in the current API functions :D

Reply:
Created #13 for that. Just ignore it for now :)